src/main/java/sg/dex/starfish/DataAsset.java

```
@@ -61,11 +73,71 @@ public default boolean isDataAsset() {
 
 		return buffer.toByteArray();
 	}
-	
+
 	/**
 	 * Gets the size of this data asset's content
+	 *
 	 * @return The size of the asset in bytes
 	 */
 	public abstract long getContentSize();
 
+
+	/**
+	 * This method is to validate the hash of the asset content .
+	 * it will calculate the hash of the content of an Asset
+	 * then compare with the hash value included in metadata,
+	 * if both are not the same , StarfishValidation Exception will be thrown
+	 *
+	 * @throws StarfishValidationException if hash content is not matched , exception will be thrown
+	 */
+	public default void validateContentHash() {
+
+		String contentHashFromMetadata = (String) getMetadata().get(Constant.CONTENT_HASH);
+		String contentHash = Hex.toString(Hash.keccak256(Utils.stringFromStream(this.getContentStream())));
+		if (null != contentHashFromMetadata && !contentHashFromMetadata.equals(contentHash)) {
```

Should throw an exception with no content hash in metadata. Check for this first.

---

resolved in latest PR